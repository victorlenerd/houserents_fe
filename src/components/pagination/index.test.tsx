import * as React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Pagination from "./index";

test("Pagination", () => {
    const onChangeHandler = jest.fn(page => page);

    const { asFragment, getByTestId } = render(<Pagination total={50} itemsPerPage={10} onChange={onChangeHandler} />);

    expect(asFragment()).toMatchSnapshot();

    const select = getByTestId("pagination-select");

    fireEvent.change(select, {
       target: {
           value: 2
       }
    });

    expect(onChangeHandler).toBeCalledWith(2);
});