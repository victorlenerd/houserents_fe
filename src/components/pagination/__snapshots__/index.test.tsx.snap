// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination 1`] = `
<DocumentFragment>
  <div
    class="pagination-container"
  >
    <select
      data-testid="pagination-select"
    >
      <option
        value="1"
      >
        1
      </option>
      <option
        value="2"
      >
        2
      </option>
      <option
        value="3"
      >
        3
      </option>
      <option
        value="4"
      >
        4
      </option>
      <option
        value="5"
      >
        5
      </option>
    </select>
    <div>
      Of 5
    </div>
  </div>
</DocumentFragment>
`;
